github/client.go

```diff
@@ -723,6 +723,20 @@ func (client *Client) FindOrCreateToken(user, password, twoFactorCode string) (t
 		} else {
 			errInfo, e := res.ErrorInfo()
 			if e == nil {
+				// If we got a un/pw message the user may have provided an
+				// access token rather than a password. Test if it's valid,
+				// and return it if it is
+				if strings.TrimSpace(errInfo.Message) == "This API can only be accessed with username and password Basic Auth" {
```

Comment:
Best not to tie this behavior to the exact language in the error message of the server response. That message could change.

How about a totally different approach? If a password of 40 or more characters was pasted, first check directly if it's a valid token (by requesting `api.github.com/user` or similar), otherwise fall back to the basic auth dance?

Reply:
I actually was going to check on any 40-character password until I noticed that github already throws this particular error only when you give it an oauth token. But you're right, it does suck to depend on a string, I wish there was a more reliable message in this case!

I'll change it to your idea.